packages/kcms/src/match/service/generatePre.test.ts

```diff
@@ -36,18 +41,94 @@ describe('GeneratePreMatchService', () => {
     ],
   ];
 
-  it('正しく予選対戦表を生成できる', async () => {
-    const generated = await generateService.handle('elementary');
+  it('正しく予選対戦表を生成できる - 部門ごと', async () => {
+    const { generateService, preMatchRepository } = createGenerateService([
+      ...testTeamData.values(),
+    ]);
+
+    const generated = await generateService.generateByDepartment('elementary');
     expect(Result.isOk(generated)).toBe(true);
     const res = Result.unwrap(generated);
 
     for (let i = 0; i < config.match.pre.course.elementary.length; i++) {
       const course = res.filter((v) => v.getCourseIndex() === i + 1);
-      const pair = course.map((v) => [
+      const pairs = course.map((v) => [
         testTeamData.get(v.getTeamID1() ?? ('' as TeamID))?.getTeamName(),
         testTeamData.get(v.getTeamID2() ?? ('' as TeamID))?.getTeamName(),
       ]);
-      expect(pair).toStrictEqual(expectedTeamPair[i]);
+      expect(pairs).toStrictEqual(expectedTeamPair[i]);
+    }
+
+    await Promise.all(
+      res.map(async (createdMatch) => {
+        const res = await preMatchRepository.findByID(createdMatch.getID());
+        expect(res).toSatisfy(Option.isSome);
+
+        const match = Option.unwrap(res);
+        expect(match).toStrictEqual(createdMatch);
+      })
+    );
+  });
+
+  it('正しく予選対戦表を生成できる - すべての部門', async () => {
+    const { generateService, preMatchRepository } = createGenerateService([
+      ...testTeamData.values(),
+    ]);
+
+    const generated = await generateService.generateAll();
+    expect(Result.isOk(generated)).toBe(true);
+    const res = Result.unwrap(generated);
+
+    for (const [departmentType, createdMatches] of res) {
+      for (let i = 0; i < config.match.pre.course[departmentType].length; i++) {
+        const course = createdMatches!.filter((v) => v.getCourseIndex() === i + 1);
+        const pairs = course.map((v) => [
+          testTeamData.get(v.getTeamID1() ?? ('' as TeamID))?.getTeamName(),
+          testTeamData.get(v.getTeamID2() ?? ('' as TeamID))?.getTeamName(),
+        ]);
+        expect(pairs).toStrictEqual(expectedTeamPair[i]);
+      }
+    }
+
+    const allMatches = [...res.values()].flat();
+
+    await Promise.all(
+      allMatches.map(async (createdMatch) => {
+        createdMatch.getMatchIndex();
+        const res = await preMatchRepository.findByID(createdMatch.getID());
+        expect(res).toSatisfy(Option.isSome);
+
+        const match = Option.unwrap(res);
+        expect(match).toStrictEqual(createdMatch);
+      })
+    );
+
+    const matchesByCourse = new Map<number, PreMatch[]>();
+    for (const match of allMatches) {
+      const courseIndex = match.getCourseIndex();
+      if (!matchesByCourse.has(courseIndex)) {
+        matchesByCourse.set(courseIndex, []);
+      }
+      matchesByCourse.get(courseIndex)!.push(match);
```

Comment:
できるだけ non-null assertion は使いたくないので↓のような感じでお願いします :pray:
```suggestion
      const matches = matchesByCourse.get(courseIndex) ?? []
      matches.push(match)
      if (!matchesByCourse.has(courseIndex)) {
        matchesByCourse.set(courseIndex, matches);
      }
```